tests/cloudpickle_test.py

```diff
@@ -1802,6 +1802,73 @@ def g(x):
 
         self.assertEqual(f2.__annotations__, f.__annotations__)
 
+    @unittest.skipIf(sys.version_info >= (3, 7),
+                     "pickling annotations is supported starting Python 3.7")
+    def test_function_annotations_silent_dropping(self):
+        # Because of limitations of typing module, cloudpickle does not pickle
+        # the type annotations of a dynamic function or class for Python < 3.7
+
+        class UnpicklableAnnotation:
+            # Mock Annotation metaclass that errors out loudly if we try to
+            # pickle one of its instances
+            def __reduce__(self):
+                raise Exception("not picklable")
+
+        unpickleable_annotation = UnpicklableAnnotation()
+
+        def f(a: unpickleable_annotation):
+            return a
+
+        with pytest.raises(Exception):
+            cloudpickle.dumps(f.__annotations__)
+
+        depickled_f = pickle_depickle(f, protocol=self.protocol)
+        assert depickled_f.__annotations__ == {}
+
+    @unittest.skipIf(sys.version_info >= (3, 7) or sys.version_info < (3, 6),
```

Comment:
This condition is really hard to parse. It should be equivalent to:

```python
@unittest.skipIf(sys.version_info[:2] != (3, 6))
```

but then contratdicts the message "pickling annotations is supported starting Python 3.7" because the test is not skipped and passes on Python 3.5.

Reply:
The tests actually test that annotations are properly dropped on Python 3.6.